nimble/controller/src/ble_ll_adv.c

```diff
@@ -2564,6 +2576,24 @@ ble_ll_adv_reschedule_periodic_event(struct ble_ll_adv_sm *advsm)
     ble_ll_adv_sync_schedule(advsm, false);
 }
 
+static void
+ble_ll_adv_update_did_periodic(struct ble_ll_adv_sm *advsm)
```

Comment:
instead of duplicating code refactor old code to simply return new value instead of updating struct, i.e.

static uint16_t ble_ll_adv_adi_update_did(uint16_t adi) {}

...

advsm->adv_adi = ble_ll_adv_adi_update_did(advsm->adv_adi)